lib/adapters/command-execution-adapter.ts

```diff
@@ -0,0 +1,26 @@
+import { ExecuteCommandParams } from "../languageclient";
+import { LanguageClientConnection } from "../main";
+
+export type CommandCustomCallbackFunction = (command: ExecuteCommandParams) => Promise<any | void>;
+
+export default class CommandExecutionAdapter {
+    private static commandsCustomCallbacks: Map<string, CommandCustomCallbackFunction> = new Map<string, CommandCustomCallbackFunction>();
+
+    public static registerCustomCallbackForCommand(command: string, callback: CommandCustomCallbackFunction): void {
```

Comment:
How are we exposing this function to the package author?

Reply:
Come to think about it, I just assumed one would invoke that method directly via the `CommandExecutionAdapter` static class, but I guess we could also have a method in `AutoLanguageClient` that simply calls that adapter. What'd you had in mind?